### PR TITLE
Add Boba predeployed contracts

### PR DESCRIPTION
**Pull request by boyuan-chen**
This PR removes the GovernanceToken deployed at `0x4200000000000000000000000000000000000042` and adds Boba redeployed contracts.

- The proxy contracts are `0x4200000000000000000000000000000000000020` (Proxy__BobaTuringCredit) and `0x4200000000000000000000000000000000000024` (Proxy__BobaGasPriceOracle). The bytecode is replace by the [proxy](https://github.com/bobanetwork/v3-anchorage/blob/boba-develop/packages/contracts-bedrock/contracts/universal/Proxy.sol) contract. The old legacy proxyTarget and proxyOwner slots are deleted. Other slots are not touched.
- The implementation contracts at `0x4200000000000000000000000000000000000021`(BobaTuringCredit), `0x4200000000000000000000000000000000000022`(BobaTuringHelper) and `0x4200000000000000000000000000000000000025` (BobaGasPriceOracle) are destroyed and replaced by the [proxy](https://github.com/bobanetwork/v3-anchorage/blob/boba-develop/packages/contracts-bedrock/contracts/universal/Proxy.sol) contract. The storages are all deleted and the new admin slot is added.
- The BOBA L2 token contract are not touched. `0x4200000000000000000000000000000000000023`

### CODE REVIEW

File: boba-chain-ops/genesis/check.go [hunk 1]
```diff
@@ -174,6 +191,59 @@ func PostCheckMigratedDB(
...
+			}
+			ExpectedStorageSlots[addr][key] = val
+		}
+	}
+	return nil
+}
+
+// PostCheckBobaLegacyProxyImplementation will check that the legacy proxy implementation
+// contracts are replaced by the proxy contract and storage are wiped out except the admin slot.
+func PostCheckBobaLegacyProxyImplementation(tx kv.Tx) error {
```
Comment (souradeep-das) [inline boba-chain-ops/genesis/check.go:213]: can you help me understand this -
I see the legacy implementation contracts are deleted (0x4200000000000000000000000000000000000021), with storage except the admin slot. why do we delete this? the implementation contracts shouldn't already have storage slots set (including an admin slot which should be on the proxy)? do we want to redeploy new implementations? what are the legacy implementation contracts replaced by?

File: boba-chain-ops/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+2. A `genesis.json` file that contains the genesis block configuration.
+3. A list of addresses that transacted on the network prior to this past regenesis.
+4. A list of addresses that performed approvals on prior versions of the OVM ETH contract.
+5. A list of msg information that stores the cross domain message from L2 to L1
+
+It creates an initialized Bedrock erigon database as output. It does this by performing the following steps:
+
+1. Create genesis (types.Genesis) from `genesis.json` and `allocation.json` in memory.
+2. Set up the proxy contract for the predeploy contracts and update the bytecode of the predeploy contracts.
+3. Migrate Boba legacy proxy implementation contract the new proxy contract and delete the slots
```
Comment (souradeep-das) [inline boba-chain-ops/README.md:16]: when you mean migrate and delete slots, you mean on the new proxy contract right?

File: boba-chain-ops/genesis/check.go (hunk 1 above)
Comment (boyuan-chen) [reply boba-chain-ops/genesis/check.go:213]: Yes, the new implementation contracts are redeployed to a 0xcd... address. The legacy implementation contracts are replaced by proxy contract with the admin slot. https://github.com/bobanetwork/v3-anchorage/blob/boba-develop/packages/contracts-bedrock/contracts/universal/Proxy.sol
We can set the implementation contract to our own contract to use this proxy address.

File: boba-chain-ops/README.md (hunk 2 above)
Comment (boyuan-chen) [reply boba-chain-ops/README.md:16]: The slots of our old implementation contracts are deleted. This means that the slots of `0x4200000000000000000000000000000000000021`, `0x4200000000000000000000000000000000000022` and `0x4200000000000000000000000000000000000025` are deleted. The bytecode is replaced by the proxy contract and the admin slot is added.

File: boba-chain-ops/genesis/check.go (hunk 1 above)
Comment (boyuan-chen) [reply boba-chain-ops/genesis/check.go:213]: Optimism generates 2048 proxy contracts for the future usage. 
https://github.com/bobanetwork/v3-anchorage/blob/16e8aeb0c2e5fb11ca7aabc592f2f5cdc40e77df/op-chain-ops/genesis/setters.go#L78

We can still use `0x4200000000000000000000000000000000000021` for the implementation contract if we want to.

Comment (souradeep-das) [reply boba-chain-ops/genesis/check.go:213]: ok got it, so the new implementation will be on the 0xcd.. address

**Pull request merged**